File: src/main/java/com/sliusar/todolist/services/ScheduleService.java
package com.sliusar.todolist.services;

import com.sliusar.todolist.models.Schedule;
import com.sliusar.todolist.models.Todo;

import java.util.List;
import java.util.Optional;

public interface ScheduleService {
    List<Schedule> getAll();
    Optional<Schedule> getById(int id);
    void save(Schedule schedule);
    void update(Schedule schedule, int id);
    void deleteById(int id);



}

File: src/main/frontend/src/app/services/schedule.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Schedule} from "../models/schedule";
import {Todo} from "../models/todo";

@Injectable({
  providedIn: 'root'
})
export class ScheduleService {
  URL = "http://localhost:8080/schedules"
  constructor(private httpClient: HttpClient) { }


  getAll(): Observable<any>{
    return this.httpClient.get<Schedule[]>(this.URL)
  }

  getById(id: number): Observable<any>{
    return this.httpClient.get<Schedule>(`${this.URL}/${id}`)
  }

  save(schedule: Schedule): Observable<any> {
    return this.httpClient.post<Schedule>(`${this.URL}/save`, schedule)
  }

  update(schedule: Schedule, id: number): Observable<any>{
    return this.httpClient.put<Schedule>(`${this.URL}/${id}/update`, schedule)
  }

  deleteById(id: number): Observable<any>{
    return this.httpClient.delete<number>(`${this.URL}/${id}/delete`);
  }

  updateTodo(todo: Todo, scheduleId: any){
    return this.httpClient.put<Todo>(`${this.URL}/${scheduleId}/update/${todo.id}`, todo)
  }
}
